handlers/admin/datamanager/admin_dashboard.cfc

```diff
@@ -0,0 +1,231 @@
+				, icon       = "fa-clone"
+				, class      = ""
+				, contextKey = "c"
+			} );
+		} else {
+			actions.append( '<a class="disabled"><i class="fa fa-fw fa-clone light-grey"></i></a>' );
+		}
+		if ( isTrue( record.canDelete ) ) {
+			actions.append( {
+				  link       = event.buildAdminLink( objectName=objectName, recordid=recordId, operation="deleteRecordAction" )
+				, icon       = "fa-trash-o"
+				, class      = "confirmation-prompt"
+				, title      = translateResource( uri="cms:datamanager.deleteRecord.prompt", data=[ prc.objectTitle, record.name ] )
+				, contextKey = "d"
+			} );
+		} else {
+			actions.append( '<a class="disabled"><i class="fa fa-fw fa-trash-o light-grey"></i></a>' );
+		}
+
+		return actions;
+	}
+
+	private void function extraTopRightButtonsForViewRecord( event, rc, prc, args={} ) {
+		var objectName = args.objectName ?: "";
+		var recordId   = prc.recordId    ?: "";
+		var record     = prc.record      ?: {};
+
+		args.actions   = args.actions    ?: [];
+
+		if ( !adminDashboardService.userCanEditDashboard( event.getAdminUserId(), recordId ) ) {
+			return;
+		}
+
+		args.actions.prepend( {
+			  link      = event.buildAdminLink( objectName=objectName, operation="sharing", recordId=recordId )
+			, btnClass  = "btn-default"
+			, iconClass = "fa-users"
+			, title     = translateResource( "preside-objects.admin_dashboard:sharing.btn" )
+		} );
+	}
+
+	public void function sharing() {
+		var recordId = rc.id ?: "";
+
+		event.initializeDatamanagerPage(
+			  objectName = "admin_dashboard"
+			, recordId   = recordId
+		);
+
+		if ( !adminDashboardService.userCanEditDashboard( event.getAdminUserId(), recordId ) ) {
+			event.accessDenied();
+		}
+
+		event.addAdminBreadCrumb(
+			  title = translateResource( "preside-objects.admin_dashboard:sharing.breadcrumb.title" )
+			, link  = ""
+		);
+
+		prc.savedData    = queryGetRow( prc.record, prc.record.recordcount );
+		prc.formName     = "preside-objects.admin_dashboard.sharing";
+		prc.pageTitle    = translateResource( "preside-objects.admin_dashboard:sharing.page.title" );
+		prc.pageSubTitle = translateResource( "preside-objects.admin_dashboard:sharing.page.subtitle" );
+	}
+
+	private string function buildSharingLink( event, rc, prc, args={} ) {
+		var qs = "id=#( args.recordId ?: "" )#";
+
+		if ( Len( Trim( args.queryString ?: "" ) ) ) {
+			qs &= "&#args.queryString#";
+		}
+
+		return event.buildAdminLink( linkto="datamanager.admin_dashboard.sharing", querystring=qs );
+	}
+
+	public void function sharingAction( event, rc, prc, args={} ) {
+		var object           = "admin_dashboard";
+		var formName         = "preside-objects.admin_dashboard.sharing";
+		var formData         = event.getCollectionForForm( formName );
+		var recordId         = rc.id ?: "";
+		var validationResult = "";
+		var persist          = "";
+
+		if ( !adminDashboardService.userCanEditDashboard( event.getAdminUserId(), recordId ) ) {
+			event.accessDenied();
+		}
+
```

Comment:
Rather than all the logic below, can do like:

```cfc
runEvent( event="admin.datamanager._editRecordAction", prepostExempt=true, private=true, args={
    // ...
    , formName = "preside-objects.admin_dashboard.sharing"
    , audit = true
    , auditAction = "edit_sharing_options"
} );
```

^^ (see method for other args you'll need/want). This will give you most of the code below + also ensure audit trail entry is added.

Alternatively, can/should manually add admin audit log entry for the sharing update.